rusqlite_migration/src/lib.rs

```diff
@@ -600,13 +600,19 @@ impl<'m> Migrations<'m> {
         res
     }
 
-    /// Maximum version defined in the migration set
-    #[allow(clippy::missing_panics_doc)]
+    /// Return the maximum version defined in the migration set.
+    ///
+    /// If you are using [`Migrations::from_directory()`] and `include_dir!`, it is non-trivial to
+    /// access the number of migrations loaded, so you may need to use this method.
+    ///
+    /// For the most common scenarios though, you should be able to just call
+    /// [`Migrations::to_latest`].
     pub fn max_schema_version(&self) -> SchemaVersion {
         match self.ms.len() {
             0 => SchemaVersion::NoneSet,
             v => SchemaVersion::Inside(
-                NonZeroUsize::new(v).expect("Already checked for 0 in previous match arm"),
+                NonZeroUsize::new(v)
+                    .unwrap_or_else(|| unreachable!("Already checked for 0 in previous match arm")),
```

Comment:
Your version avoids the overly broad `#[allow]`, so is much better.

FYI: I tried to apply the `#[allow]` to just the `expect()` expression; `rustc` understands that, but unfortunately it seems `clippy` does not.

Nice!